ONSdigital/census-rm-ops
# Add page to generate unaddressed QID batch

# Motivation and Context
We have a tool for generating unaddressed QID batches by a script: [census-rm-qid-batch-runner](https://github.com/ONSdigital/census-rm-qid-batch-runner).
It is convenient to have an ops dev tool UI page to do the same, the basic code has been copied over and a simple page added linked to from each action plan page to upload and run a QID batch config CSV. Note this is run synchronously in the HTTP post request so may time out if the batch is too large.

# What has changed
- Add code from census-rm-qid-batch-runner
- Add generate qid batch page
- Add link to generate qid batch page on each action plan page
- Add config var for new queue name
- Add unit tests
- Small refactor to share code with sample loader 
- Always declare queue in rabbit context

# How to test?
Run the tool and create an action plan. On the action plan page, click the link to generate qid batch. Upload a valid QID batch config CSV, then check the request messages have been sent to the correct queue

# Links
https://trello.com/c/zRjY6YpP/714-unaddressed-batch-runner-8
https://github.com/ONSdigital/census-rm-qid-batch-runner
